ngsPETSc/utils/firedrake/meshes.py

```diff
@@ -70,105 +76,156 @@ def refineMarkedElements(self, mark):
     else:
         raise NotImplementedError("No implementation for dimension other than 2 and 3.")
 
+
+def _slow_cdist(XA, XB):
+    dist = np.zeros([len(XA), len(XB)])
+    for ii, a in enumerate(XA):
+        for jj, b in enumerate(XB):
+            dist[ii, jj] = np.linalg.norm(b - a)
+    return dist
+
+
+if not HAVE_SCIPY:
+    cdist = PETSc.Log.EventDecorator()(_slow_cdist)
+
+
+@PETSc.Log.EventDecorator()
```

Comment:
What does `PETSc.Log.EventDecorator` do ?

Reply:
Adds events to the flamegraph and the PETSc log view. It's very useful for profiling

Reply:
![jack](https://github.com/user-attachments/assets/220b6356-e18a-4ba7-a565-ae5dcc9edbf2)